src/canopy_calcs.F90

```diff
@@ -325,20 +325,87 @@ SUBROUTINE canopy_calcs(nn)
-                                tleaf_ave24_tmp_3d(nn,i,j,:)    = tleaf_ave
-                                ppfd_sun240_tmp_3d(nn,i,j,:)    = ppfd_sun
-                                ppfd_shade240_tmp_3d(nn,i,j,:)  = ppfd_shade
-                                tleaf_sun240_tmp_3d(nn,i,j,:)   = tleaf_sun
-                                tleaf_shade240_tmp_3d(nn,i,j,:) = tleaf_shade
-                                tleaf_ave240_tmp_3d(nn,i,j,:)   = tleaf_ave
-                                !Track times for daily maximum/minimum temperature and maximum wind speed calculation below
-                                tmp2mref_tmp_3d(nn,i,j)         = tmp2mref
-                                ubzref_tmp_3d(nn,i,j)           = ubzref
+
+                                if (nn .le. 24) then
+                                    !Track times for moving time window (running) average below
+                                    ppfd_sun24_tmp_3d(nn,i,j,:)     = ppfd_sun
+                                    ppfd_shade24_tmp_3d(nn,i,j,:)   = ppfd_shade
+                                    tleaf_sun24_tmp_3d(nn,i,j,:)    = tleaf_sun
+                                    tleaf_shade24_tmp_3d(nn,i,j,:)  = tleaf_shade
+                                    tleaf_ave24_tmp_3d(nn,i,j,:)    = tleaf_ave
+                                    ppfd_sun240_tmp_3d(nn,i,j,:)    = ppfd_sun
+                                    ppfd_shade240_tmp_3d(nn,i,j,:)  = ppfd_shade
+                                    tleaf_sun240_tmp_3d(nn,i,j,:)   = tleaf_sun
+                                    tleaf_shade240_tmp_3d(nn,i,j,:) = tleaf_shade
+                                    tleaf_ave240_tmp_3d(nn,i,j,:)   = tleaf_ave
+                                    tmp2mref_tmp_3d(nn,i,j)         = tmp2mref
+                                    ubzref_tmp_3d(nn,i,j)           = ubzref
+                                else if (nn .le. 240) then
+                                    ppfd_sun24_tmp_3d(25,i,j,:)     = ppfd_sun
+                                    ppfd_shade24_tmp_3d(25,i,j,:)   = ppfd_shade
+                                    tleaf_sun24_tmp_3d(25,i,j,:)    = tleaf_sun
+                                    tleaf_shade24_tmp_3d(25,i,j,:)  = tleaf_shade
+                                    tleaf_ave24_tmp_3d(25,i,j,:)    = tleaf_ave
+                                    ppfd_sun240_tmp_3d(nn,i,j,:)    = ppfd_sun
+                                    ppfd_shade240_tmp_3d(nn,i,j,:)  = ppfd_shade
+                                    tleaf_sun240_tmp_3d(nn,i,j,:)   = tleaf_sun
+                                    tleaf_shade240_tmp_3d(nn,i,j,:) = tleaf_shade
+                                    tleaf_ave240_tmp_3d(nn,i,j,:)   = tleaf_ave
+                                    tmp2mref_tmp_3d(25,i,j)         = tmp2mref
+                                    ubzref_tmp_3d(25,i,j)           = ubzref
+                                    ppfd_sun24_rst_3d(:,i,j,:)     = ppfd_sun24_tmp_3d(1:24,i,j,:)
```

Comment:
So, no functionality yet to read in "rst" variables from external restart files?  See earlier comments on how these restart variables could look like too.  Maybe we need to chat as not sure how these work right now.